source/common/common/assert.h

```diff
@@ -156,10 +162,20 @@ void resetEnvoyBugCountersForTest();
 // _ASSERT_VERBOSE, and this will call _ASSERT_VERBOSE,(__VA_ARGS__)
 #define ASSERT(...)                                                                                \
   EXPAND(_ASSERT_SELECTOR(__VA_ARGS__, _ASSERT_VERBOSE, _ASSERT_ORIGINAL)(__VA_ARGS__))
+
+#if !defined(NDEBUG) || defined(ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE)
+// debug build or all ASSERTs compiled in release.
+#define SLOW_ASSERT(...) ASSERT(__VA_ARGS__)
+#else
+// Non-implementation of SLOW_ASSERTs when building only ENVOY_LOG_FAST_DEBUG_ASSERT_IN_RELEASE.
+#define SLOW_ASSERT _NULL_ASSERT_IMPL
+#endif
+
 #else
 #define ASSERT _NULL_ASSERT_IMPL
 #define KNOWN_ISSUE_ASSERT _NULL_ASSERT_IMPL
-#endif // !defined(NDEBUG) || defined(ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE)
+#define SLOW_ASSERT _NULL_ASSERT_IMPL
+#endif // !defined(NDEBUG) || defined(ENVOY_LOG_FAST_DEBUG_ASSERT_IN_RELEASE)
```

Comment:
I feel like I must be missing something, but this looks like the #if/#else/#endif blocks are unbalanced:

```
#if !defined(NDEBUG) || defined(ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE)
  ...
#else
  ...
#endif

#else
  ...
#endif // !defined(NDEBUG) || defined(ENVOY_LOG_FAST_DEBUG_ASSERT_IN_RELEASE)
```

I don't see where the `#if` for the `#else` at line 174 is.

Reply:
It's actually at line 127. LIne 127-174 is the whole if D:

Reply:
Ah. There it is. Should the comment on 178 include `|| defined(ENVOY_LOG_FAST_DEBUG_ASSERT_IN_RELEASE)`?

Reply:
Or no `defined(ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE)`. But what I'm getting at is that the endif comment has 2 terms and the actual if has 3.

Reply:
yes! you're right about the comment. thanks.